src/main/scala/com/koverse/example/spark/NaiveBayesPredictTransform.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2016 Koverse, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.koverse.example.spark
+
+import java.lang
+
+import com.koverse.com.google.common.collect.Lists
+import com.koverse.sdk.Version
+import com.koverse.sdk.data.{Parameter, SimpleRecord}
+import com.koverse.sdk.transform.spark.{JavaSparkTransform, JavaSparkTransformContext}
+import com.koverse.sdk.transform.spark.sql.KoverseSparkSql
+import org.apache.spark.api.java.JavaRDD
+import org.apache.spark.ml.feature.{HashingTF, Tokenizer}
+import org.apache.spark.mllib.classification.NaiveBayesModel
+import org.apache.spark.mllib.linalg.{SparseVector, Vectors}
+import org.apache.spark.mllib.regression.LabeledPoint
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.DataFrame
+
+class NaiveBayesPredictTransform extends JavaSparkTransform {
+
+  override def execute(context: JavaSparkTransformContext): JavaRDD[SimpleRecord] = {
+
+    val SQLContext = KoverseSparkSql.createSqlContext(context.getJavaSparkContext.sc)
+    val inputCollectionId = context.getInputCollectionIds().get(0)
+    val inputDataFrame:DataFrame = KoverseSparkSql.createDataFrame(context.getInputCollectionRdds().get(inputCollectionId),
+      SQLContext , context.getInputCollectionSchemas().get(inputCollectionId))
+
+    val modelId = context.getInputCollectionIds().get(1)
+    val modelDataFrame:DataFrame = KoverseSparkSql.createDataFrame(context.getInputCollectionRdds().get(modelId),
```

Comment:
Would it be easier to just leave this as an RDD of SimpleRecords  and then just collect it below? It doesn't seem like we benefit from having a Dataframe for the model Data Set.

Reply:
This is to use the Tokinizer, it takes only  DataFrames , https://spark.apache.org/docs/1.6.3/ml-features.html#tokenizer